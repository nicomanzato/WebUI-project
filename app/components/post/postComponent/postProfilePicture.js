import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  Text,
  Image,
  TouchableHighlight
} from 'react-native';

export const PostProfilePicture = (props) => {
  return (
    <View style={styles.profileView}>
      <TouchableHighlight
        underlayColor='#dddddd'
        onPress={props.onProfilePicPress}
      >
        <Image
          fadeDuration={0}
          source={{uri: props.user.profile_image_url_https}}
          style={styles.profilePic}/>
      </TouchableHighlight>
    </View>
  );
}

const styles = StyleSheet.create({
  profileView: {
    flex: 0.2,
  },

  profilePic: {
    resizeMode: 'contain',
    alignSelf: 'center',
    height: 50,
    width: 50,
    borderRadius: 25,
  },

});
