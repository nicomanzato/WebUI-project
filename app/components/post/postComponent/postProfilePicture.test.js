import React from 'react'
import {PostProfilePicture} from './postProfilePicture'
import PostMock2 from './../../../store/post/mock/postMock2'
import { shallow, mount, ReactWrapper } from 'enzyme'

describe('post component', () => {

  it('should render correctly', () => {
    const wrapper = shallow(
      <PostProfilePicture user={PostMock2.user} />
    );

    expect(wrapper).toMatchSnapshot();
  });

});
