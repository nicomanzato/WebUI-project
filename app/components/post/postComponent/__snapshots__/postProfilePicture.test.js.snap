// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post component should render correctly 1`] = `
<Component
  style={
    Object {
      "flex": 0.2,
    }
  }
>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    underlayColor="#dddddd"
  >
    <Image
      fadeDuration={0}
      source={
        Object {
          "uri": "https://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "borderRadius": 25,
          "height": 50,
          "resizeMode": "contain",
          "width": 50,
        }
      }
    />
  </TouchableHighlight>
</Component>
`;
